fastlane/lanes/test.rb

```diff
@@ -43,6 +43,49 @@
       UI.test_failure!(message)
     end
   end
+
+  #####################################################################################
+  # build_and_run_jetpack_instrumented_test
+  # -----------------------------------------------------------------------------------
+  # Run instrumented tests in Google Firebase Test Lab
+  # -----------------------------------------------------------------------------------
+  # Usage:
+  # bundle exec fastlane build_and_run_jetpack_instrumented_test
+  #
+  #####################################################################################
+  desc "Build the application and instrumented tests, then run the tests in Firebase Test Lab"
+  lane :build_and_run_jetpack_instrumented_test do | options |
+    gradle(tasks: ['WordPress:assembleJetpackVanillaDebug', 'WordPress:assembleJetpackVanillaDebugAndroidTest'])
+
+    # Run the instrumented tests in Firebase Test Lab
+    firebase_login(
+      key_file: GOOGLE_FIREBASE_SECRETS_PATH
+    )
+
+    apk_dir = File.join(PROJECT_ROOT_FOLDER, 'WordPress', 'build', 'outputs', 'apk')
+
+    test_succeeded = android_firebase_test(
+      project_id: firebase_secret(name: 'project_id'),
+      key_file: GOOGLE_FIREBASE_SECRETS_PATH,
+      model: 'Pixel2.arm',
+      version: 30,
+      test_apk_path: File.join(apk_dir, 'androidTest', 'jetpackVanilla', 'debug', 'org.wordpress.android-jetpack-vanilla-debug-androidTest.apk'),
+      apk_path: File.join(apk_dir, 'jetpackVanilla', 'debug', 'org.wordpress.android-jetpack-vanilla-debug.apk'),
+      test_targets: 'package org.wordpress.android.e2e',
```

Comment:
For Jetpack, only targetting UI tests under the `org.wordpress.android.e2e` package

Reply:
👋 @jostnes !

Why is that so, why are we targeting only those [15 test cases](https://console.firebase.google.com/project/api-project-108380595987/testlab/histories/bh.3ce77db975c1f0c9/matrices/8732295795092882555/executions/bs.6507729498e793b9/test-cases) for Jetpack, with one of these skipped anyway? 🤔 

PS: For WordPress we are targeting [121 test cases](https://console.firebase.google.com/project/api-project-108380595987/testlab/histories/bh.b3d1e88d8b09c8be/matrices/8018251029287387893/executions/bs.bfc10ce437d81849/test-cases) overall, with the same one being skipped (just like for Jetpack above).

Reply:
My two cents is that these additional 106 instrumentation tests run for only 13 seconds.

> Since the codebase is shared and there are no visual differences between the two, I don't think there's value in including those tests, at least in this case.

The same logic can be applied to UI tests, then 🤔 I see value in running them all, especially since they are fast, _and_ we save ourselves from accidentally forgetting to include them back again later, when there are more differences between the apps.

Reply:
@jostnes I do agree with @pachlava on that. 💯 

Actually, even if they would have caused minutes of increased test run, I think it is better to start with the full suite of tests to be run on both, the `Jetpack` and `WordPress` apps, and maybe then, take it from there, if need be, when features start getting removed from `WordPress`, and thus try to separate those somehow to gain in performance of those tests completing.

For now, I would go with consistency just so that we can verify that every test that can be run on `Jetpack` is able to run and is as operational as possible.

Reply:
I've checked those tests outside of the `e2e` package.
- [URITest](https://github.com/wordpress-mobile/WordPress-Android/blob/ed574ea833797f71e28fb2abd060fee7d305376c/WordPress/src/androidTest/java/URITest.java) and [URLTest](https://github.com/wordpress-mobile/WordPress-Android/blob/ed574ea833797f71e28fb2abd060fee7d305376c/WordPress/src/androidTest/java/URLTest.java) are testing `java.net` classes, which doesn't make sense. I think we can remove these. They include **10** tests. This can be another issue.
- Other tests are testing features identical for WP and JP. But still, I believe we should test all for WP and JP. 
- On [UI tests](https://github.com/wordpress-mobile/WordPress-Android/tree/trunk/WordPress/src/androidTest/java/org/wordpress/android/e2e) document, it says `2. Run the tests in org.wordpress.android.e2e`. I think we should move all tests under `e2e` package, or update the document. This is just an idea for another issue.

Reply:
Thank you everyone for the input! Since there are opinions around this and I have some unanswered questions, I make the update to include the tests to be on both and WP and JP steps, change made in this commit https://github.com/wordpress-mobile/WordPress-Android/pull/17950/commits/037a16df4229980fd6bb0f300c4b0dd466f6093e. 

For points 1 and 3 that you mentioned @irfano are out of scope of this PR so I created this [issue](https://github.com/wordpress-mobile/WordPress-Android/issues/18014) so we don't forget about it.

I'd like to share my perspective about my decision to only include only the UI tests on Jetpack in case it's not clear. To be honest duration/time taken was not the main factor I considered, I was thinking more in the line of necessity. I checked out a sample of the non-UI connected tests (e.g. [testSafeToAddAuthToken1](https://github.com/wordpress-mobile/WordPress-Android/blob/8c610143de99b5dbacacf4e2b1177f3e3fe91d41/WordPress/src/androidTest/java/org/wordpress/android/util/WPUrlUtilsTest.java#L29) and [testGetHost1](https://github.com/wordpress-mobile/WordPress-Android/blob/8c610143de99b5dbacacf4e2b1177f3e3fe91d41/WordPress/src/androidTest/java/org/wordpress/android/util/UrlUtilsTest.java#L12)) and from the samples, the tests don't seem to run on an emulator (making it unaware if the app is WP or JP, which is the main difference between the two steps). Though there are questions that I still don't know the answers to:

1. I wonder if the non UI connected tests will diverge between WP and JP like how the UI tests will be in the future as we add/remove features to the apps?
2. I'm not sure if running it on Jetpack will have additional value or will it be a redundant test run (will both tests always end with the same exact results and what are the differences between it running on WP and JP steps since it doesn't run on the app)?

Since there are unanswered questions and making an assumption is probably not a good way to make this decision, making it run on both for now seems like the way to go. We can change it later if needed once we know for certain the answers to these.

Thank you again all for the input! 🙇